src/SHA.jl

```diff
@@ -51,7 +51,7 @@ for (f, ctx) in [(:sha1, :SHA1_CTX),
         # open("test.txt") do f
         #     sha256(f)
         # done
-        $f(io::IO) = $f(readbytes(io))
+        $f(io::IO) = @compat $f(read(io))
```

Comment:
I think the right syntax in that case is `read(io  UInt8, 1)` and not `Vector{UInt8}`. In the definition `read(s::IO, t::T, dims)`, I think `T` is constrained to be a bits type, as evidenced by [this line](https://github.com/JuliaLang/julia/blob/master/base/io.jl#L225). Now, when I put your code in there, I get an error like this:
```
ERROR: LoadError: MethodError: Cannot `convert` an object of type Type{Array{UInt8,1}} to an object of type Array{UInt8,N}
This may have arisen from a call to the constructor Array{UInt8,N}(...)
```
This error message could be a lot more helpful, though. 
cc: @tkelman